boa/vm/fork.py

```diff
@@ -115,19 +126,12 @@ def __init__(self, *args, **kwargs):
         rpc_kwargs = self._rpc_init_kwargs.copy()
 
         block_identifier = rpc_kwargs.pop("block_identifier", "safe")
-        self._rpc = CachingRPC.get_rpc(**rpc_kwargs)
-
-        if block_identifier not in (
-            "safe",
-            "latest",
-            "finalized",
-            "pending",
-            "earliest",
-        ):
+        self._rpc = CachingRPC(self._rpc, **rpc_kwargs)
+
+        if block_identifier not in _PREDEFINED_BLOCKS:
             block_identifier = to_hex(block_identifier)
 
-        # do not cache - use raw_fetch
-        self._block_info = self._rpc._raw_fetch_single(
+        self._block_info = self._rpc.fetch(
```

Comment:
hm i'm not sure we can make this change -- doesn't CachingRPC cache the result?

Reply:
I've added a fetch_uncached method to address your concern